nixos/modules/services/web-apps/weblate.nix

```diff
@@ -0,0 +1,421 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.weblate;
+
+  dataDir = "/var/lib/weblate";
+  staticRoot = "${dataDir}/static";
+  compressRoot = "${dataDir}/compressor-cache";
+  settingsDir = "${dataDir}/settings";
+
+  finalPackage = cfg.package.overridePythonAttrs (old: {
+    # We only support the PostgreSQL backend in this module
+    dependencies = old.dependencies ++ cfg.package.optional-dependencies.postgres;
+    # Use a settings module in dataDir, to avoid having to rebuild the package
+    # when user changes settings.
+    makeWrapperArgs = (old.makeWrapperArgs or [ ]) ++ [
+      "--set PYTHONPATH  \"${settingsDir}\""
+      "--set DJANGO_SETTINGS_MODULE \"settings\""
+    ];
```

Comment:
I think the wording is not great here. The settings aren't mutable actually, they are just symlinked into `/var/lib`, so that we don't have a `/nix/store` reference to the settings in the weblate closure. Changing the module configuration will always deploy and restart the right settings. There wouldn't be any state.

I'll refine the comment.

Reply:
Ohh I see.
1. Why are we doing the same with `staticRoot` / `compressRoot`
2. Given that we run this through uwsgi, are the wrapper env variables even getting used anywhere?

Reply:
1. I realised that `staticRoot`/`compressRoot` where remnants of before I figured out to use a `settingsDir`. I removed these!
2. Yes, when one creates the first admin user for example, using the `weblate` executable in path. In the flake, I used weblate-env for this, which is now unnecessary. Also the migration service calls weblate directly.

Reply:
How do you feel about adding another layer of wrapper instead? Something like
```
wrappedWeblate = pkgs.runCommand "wrap-weblate" { nativeBuildInputs = [ pkgs.makeWrapper ]; } ''
  mkdir -p $out/bin
  makeWrapper ${cfg.package}/bin/weblate $out/bin/weblate \
    --set PYTHONPATH "${settings_py}"
    --set DJANGO_SETTINGS_MODULE "settings"
'';
```
It trades complexity in one place for complexity somewhere else, so I'll leave the judgment call to you. Personally `tmpfiles.d` and symlinking store paths as runtime paths feels like icky state and therefore a last resort. It's not actually so bad in this case, though.

Reply:
I thought about it a bit, but I think I'd like to leave it like it is. Personally I sometimes take comfort in knowing a semi-static location for the service configuration (compare to having to peel the path out of the systemd service definition every time the config changes) and I don't like the idea of wrapping all weblate binaries so much, even though I understand that it'd totally work to wrap the binary twice.